eureka2-ext/eureka2-eureka1-rest-api/src/test/java/com/netflix/eureka2/eureka1/rest/query/Eureka2FullFetchWithDeltaViewTest.java

```diff
@@ -0,0 +1,138 @@
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import com.netflix.eureka2.testkit.data.builder.SampleInstanceInfo;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import rx.functions.Action1;
+import rx.schedulers.Schedulers;
+import rx.schedulers.TestScheduler;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author Tomasz Bak
+ */
+public class Eureka2FullFetchWithDeltaViewTest {
+
+    private static final int APPLICATION_CLUSTER_SIZE = 3;
+    private static final long REFRESH_INTERVAL = 30000;
+
+    private final TestScheduler testScheduler = Schedulers.test();
+
+    @Rule
+    public final SourcedRegistryMockResource registryMockResource = new SourcedRegistryMockResource();
+
+    private final SourcedEurekaRegistry<InstanceInfo> registry = registryMockResource.registry();
+
+    private Eureka2FullFetchWithDeltaView view;
+
+    @Before
+    public void setUp() throws Exception {
+        view = new Eureka2FullFetchWithDeltaView(registry.forInterest(Interests.forFullRegistry()), REFRESH_INTERVAL, testScheduler);
+    }
+
+    @Test
+    public void testAfterFirstBatchApplicationsAndDeltaAreAvailable() throws Exception {
+        String webAppName = registryMockResource.uploadClusterBatchToRegistry(SampleInstanceInfo.WebServer, APPLICATION_CLUSTER_SIZE);
+        RegistryFetch registryFetch = latestCopy();
+
+        Applications fullFetch = registryFetch.getApplications();
+        Applications delta = registryFetch.getDeltaChanges();
+
+        verifyApplicationsPresent(fullFetch, webAppName);
+        assertThat(delta.getRegisteredApplications().size(), is(equalTo(0)));
+
+        verifyHashCodes(fullFetch, delta);
+    }
+
+    @Test
+    public void testSecondBatchIncludedInApplicationsAndDelta() throws Exception {
+        // First batch
+        String webAppName = registryMockResource.uploadClusterBatchToRegistry(SampleInstanceInfo.WebServer, APPLICATION_CLUSTER_SIZE);
+        testScheduler.triggerActions();
+        RegistryFetch firstCopy = latestCopy();
+
+        // Second batch
+        String backendName = registryMockResource.uploadClusterBatchToRegistry(SampleInstanceInfo.Backend, APPLICATION_CLUSTER_SIZE);
+        assertSame(firstCopy, latestCopy());
+        testScheduler.advanceTimeBy(REFRESH_INTERVAL, TimeUnit.MILLISECONDS);
+
+        RegistryFetch secondCopy = latestCopy();
+
+        Applications fullFetch = secondCopy.getApplications();
+        Applications delta = secondCopy.getDeltaChanges();
+
+        verifyApplicationsPresent(fullFetch, webAppName, backendName);
+        verifyApplicationsPresent(delta, backendName);
+
+        verifyHashCodes(fullFetch, delta);
+    }
+
+    @Test
+    public void testRemovedElementsDroppedFromApplicationsAndPresentInDelta() throws Exception {
```

Comment:
Is there a modify equivalent of this test (or is that tested together with one of the current testcases?)